tests/bindings/nodejs/HawkTracerClient.test.ts

```diff
@@ -72,50 +72,83 @@ test("Adds klass_name", (done) => {
     });
 })
 
+
+test("start (resolve)", () => {
+    expect.assertions(2);
+    mockNativeClient.start = jest.fn().mockReturnValue(true);
+    const client = new HawkTracerClient("abc");
+    return client.start().then(data => {
+        expect(data).toEqual(client);
+        expect(mockNativeClient.start).toHaveBeenCalledTimes(1);
+    });
+});
+
+test("start (reject)", () => {
+    expect.assertions(2);
+    mockNativeClient.start = jest.fn().mockImplementationOnce(() => {
+        throw new Error();
+    });
+    const client = new HawkTracerClient("abc");
+    return client.start().catch(e => {
+        expect(e).toBeInstanceOf(Error);
+        expect(mockNativeClient.start).toHaveBeenCalledTimes(1);
+    })
+});
+
+test("start (retries)", () => {
+    expect.assertions(2);
+    mockNativeClient.start = jest.fn().mockReturnValue(false);
+    const client = new HawkTracerClient("abc", 0);
+    return client.start().catch(e => { 
+        expect(e).toBeInstanceOf(Error);
+        expect(mockNativeClient.start).toHaveBeenCalledTimes(1);
```

Comment:
Could we also test if it retries as many times as instructed, say 3?

Reply:
Okay, that's a fair point